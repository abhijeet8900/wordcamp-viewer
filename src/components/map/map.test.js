import { render, screen } from "@testing-library/react";
import React from "react";
import Map from "./index.js";

describe("Map tests", () => {
  const MOCK_EVENTS = [
    {
      id: 3237405,
      title: {
        rendered: "Community Building Workshop Japan",
      },
      slug: "wordcamp-whitley-bay",
      link: "https://central.wordcamp.org/wordcamps/wordcamp-gdynia-poland/",
      "Start Date (YYYY-mm-dd)": Math.floor(Date.now() / 1000),
      "End Date (YYYY-mm-dd)": Math.floor(Date.now() / 1000),
      _venue_coordinates: {
        latitude: 55.0471853,
        longitude: -1.4460773,
      },
    },
    {
      title: {
        rendered: "Rome Community Day 2023",
      },
      slug: "wordcamp-skopje-2",
      "Start Date (YYYY-mm-dd)": Math.floor(Date.now() / 1000),
      "End Date (YYYY-mm-dd)": Math.floor(Date.now() / 1000),
      "Event Timezone": "Europe/Rome",
      _venue_coordinates: {
        latitude: 41.9953811,
        longitude: 21.4757918,
      },
    },
  ];

  it("Renders Loation markers ", () => {
    render(<Map events={MOCK_EVENTS} />);
    MOCK_EVENTS.forEach((event) => {
      const marker = screen.getByTestId(`event-${event.slug}`);
      expect(marker).toBeInTheDocument();
    });
  });

  it("Snapshot test ", () => {
    const { container } = render(<Map events={MOCK_EVENTS} />);
    expect(container).toMatchSnapshot();
  });
});
