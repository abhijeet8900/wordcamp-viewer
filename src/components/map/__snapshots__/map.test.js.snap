// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map tests Snapshot test  1`] = `
<div>
  <div
    class="w-full max-w-5xl h-[720px] mx-auto"
  >
    <div
      style="width: 100%; height: 100%; margin: 0px; padding: 0px; position: relative;"
    >
      <div
        style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
      />
      <div
        style="width: 50%; height: 50%; left: 50%; top: 50%; margin: 0px; padding: 0px; position: absolute;"
      >
        <div
          style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
        >
          <div
            style="width: 0px; height: 0px; left: -223.17825015466684px; top: -156.8669952052012px; background-color: transparent; position: absolute;"
          >
            <div
              class="pin cursor-pointer"
              data-testid="event-wordcamp-whitley-bay"
              id="event-3237405"
            >
              <svg
                aria-hidden="true"
                class="pin-icon"
                height="1em"
                role="img"
                style="color: gray; font-size: 16px;"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <path
                  d="M12 11.5A2.5 2.5 0 0 1 9.5 9A2.5 2.5 0 0 1 12 6.5A2.5 2.5 0 0 1 14.5 9a2.5 2.5 0 0 1-2.5 2.5M12 2a7 7 0 0 0-7 7c0 5.25 7 13 7 13s7-7.75 7-13a7 7 0 0 0-7-7Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
          <div
            style="width: 0px; height: 0px; left: -157.97826693688899px; top: -100.37795005381531px; background-color: transparent; position: absolute;"
          >
            <div
              class="pin cursor-pointer"
              data-testid="event-wordcamp-skopje-2"
              id="event-undefined"
            >
              <svg
                aria-hidden="true"
                class="pin-icon"
                height="1em"
                role="img"
                style="color: gray; font-size: 16px;"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <path
                  d="M12 11.5A2.5 2.5 0 0 1 9.5 9A2.5 2.5 0 0 1 12 6.5A2.5 2.5 0 0 1 14.5 9a2.5 2.5 0 0 1-2.5 2.5M12 2a7 7 0 0 0-7 7c0 5.25 7 13 7 13s7-7.75 7-13a7 7 0 0 0-7-7Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
